### ISSUE TITLE
Show current branch only; Not persistent

### ISSUE THREAD

**Issue by NOYB** [gitextensions/gitextensions#10424]
### Environment

- Git Extensions 4.0.0.15569
- Build cba315df5366fca01240d93b6e5a5077fd5e6630
- Git 2.38.1.windows.1
- Microsoft Windows NT 6.3.9600.0
- .NET 6.0.11
- DPI 96dpi (no scaling)

### Issue description

View menu
Show current branch only (Ctl-Shift-U) Not persistent.
Always defaults to Show all branches (Ctl-Shift-A) when opening repository.
Worked fin in previous versions.

### Steps to reproduce

Set "Show current branch only" (Ctrl-Shift-U)
Close repository.
Open repository.
All branches being shown.

### Did this work in previous version of GitExtensions?

Yes in all recent previous versions (3....)

**Comment by gerhardol** [comment]
The change was (mostly) intentional, filters are reset when checking out branches.
There were a number of reports where users reported issues due to remaining filters, not realizing the changes were persistent. So "current branch" is handled as "filtered branches" and reset. Reflog (that is handled as a branchfilter in 4.0) is kept though. 
This is set in `FilterInfo. ResetAllFilters()` from SetGitModule, seem to be the only code path.

@gitextensions/git-extensions-team Should we revert this?

**Comment by NOYB** [comment]
Please either revert or have a persistent option available.
99% of the time I'm only looking at the current branch.  Cumbersome having to set this every time opening a repository.  Maybe Show current branch only (persistent) (Ctl-Alt-U)
Thanks

**Comment by RussKie** [comment]
I'm of the opinion that repo-filters (such as show current branch or
reflog) should not be persisted, and should be reset whenever a repo is
changed.
My 2c

**Comment by vbjay** [comment]
I can see on checkout being too much.  On repo switch/load I don't expect filters to be remembered.  I do like the reflog being remembered.  

View all/filtered/current could follow this logic upon checkout of a new branch

- If view = filtered branches
    - If branch being checked out would be filtered
        - Switch to all
    -  If not..
        - Leave filtered view setting
- If view = all or current 
    - just keep this view setting on check out

View all/filtered/current could follow this logic upon repo switch

Reset to last current or all view of current app instance( default if new app run = all), never start with filtered.  So I 

1. Open repo x
2. Set view to current
3. Open repo y
4. view = current

1. Open repo x
2. Set view to current
3. Then later set to filtered
4. Open repo y
5. view = current

Basically remember current/all ignoring filtered in current run of app.  Do not persist anywhere but memory.  

Reflog is more of an app level view and should be remembered throughout life time of app run even if switching repos.  It's not hard to toggle if they want it changed.

**Comment by gerhardol** [comment]
> If view = filtered branches
    If branch being checked out would be filtered
        Switch to all

Refs and git-log are read in parallel, So this this would require changes to an already complex loading procedure, delaying display.
But names of last filtered branches are remembered in Advanced filter (not automatically propagated to the filtered search box).
Maybe that is what @vbjay is proposing, I do not follow completely...

I propose filtered is set to All (as it is now).
I also want to keep Reflog if set, why probably Current should be kept too.
But always changing to All is fine for me.

**Comment by NOYB** [comment]
I propose that when I open a repository it returns to where/how I left it.  Whether a new session or switching between repos within same session.
Maybe an option setting.

**Comment by vbjay** [comment]
> Refs and git-log are read in parallel

Good point.  I can see the switching to all instead being ok.
> Maybe that is what @vbjay is proposing

I am talking about which view is used.  So I am not talking about the filter settings.  I am tlaking about whether filtered branches is selected.  

![image](https://user-images.githubusercontent.com/3629489/202951934-204795c0-fa1e-403b-a752-0eb7b2b7d9d1.png)
